Public/Src/Cache/ContentStore/Library/FileSystem/PassThroughFileSystem.cs

```diff
@@ -445,89 +446,86 @@ private async Task CopyFileInsideSemaphoreAsync(AbsolutePath sourcePath, Absolut
             }
         }
 
-        private Stream OpenInternal(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
+        private Stream TryOpenFile(AbsolutePath path, FileAccess accessMode, FileMode mode, FileShare share, FileOptions options, int bufferSize)
         {
-                if (OperatingSystemHelper.IsUnixOS)
-                {
-                    if (DirectoryExists(path))
-                    {
-                        throw new UnauthorizedAccessException($"Cannot open directory {path} as a file.");
-                    }
+            if (OperatingSystemHelper.IsUnixOS)
+            {
+                return TryOpenFileUnix(path, accessMode, mode, share, options, bufferSize);
+            }
 
-                    if (mode == FileMode.Open && !FileExists(path))
-                    {
-                        return null;
-                    }
+            return TryOpenFileWin(path, accessMode, mode, share, options, bufferSize);

```

Comment:
Shouldn't this be implemented at the BuildXL.Native.IO.FileUtilities level so we can share any changes across all of BXL?

Reply:
This API is very weird and returns null in some cases. The behavior is very unusual and I doubt that it should be the same in Bxl.

So in general I agree that this class should be unified with bxl but I'm not sure that this is feasible right now.

---
In reply to: [285679929](https://github.com/microsoft/BuildXL/pull/305#discussion_r285679929) [](ancestors = 285679929)